### ISSUE TITLE
In-Game: typeof(functionName) returns "number" when the function was defined in an Object event

### ISSUE THREAD

**Issue by tinkerer-red** [YoYoGames/GameMaker-Bugs#7933]
### Description

Copy the code below and place it into a create event.

![image](https://github.com/user-attachments/assets/e33139ff-905c-498e-ba3b-fa49fa0c0935)

Russel mentioned the following ::
> I will take a look at what is happening in there... I think it is because they are not global but part of the object context
> can you file an issue (so it does not get lost)
> and I will investigate why
> it is not intentional it will be just how it is looked up

### Expected Change

should be a reference, not a number

### Steps To Reproduce

```
function foo(){}

log(foo)
log(typeof(foo))
log(typeof(method({}, foo)))
log("\n")

log(log)
log(typeof(log))
log(typeof(method({}, log)))
log("\n")

log(Script1)
log(typeof(Script1))
log(typeof(method({}, Script1)))
log("\n")
```

### How reliably can you recreate this issue using your steps above?

Always

### Which version of GameMaker are you reporting this issue for?

2024.8.1 (Monthly)

### Which platform(s) are you seeing the problem on?

Windows

### Sample Package Attached?

- [X] I have attached my bug package

### Sample Project Added?

- [X] I have included a small sample project

**Comment by rwkay** [comment]
Fixed in 2024.11

**Comment by Emc1923** [comment]
Verified as of  IDE v2024.1100.0.643  Runtime v2024.1100.0.664.
